graylog2-server/src/main/java/org/graylog2/indexer/messages/Messages.java

```diff
@@ -78,7 +78,7 @@
     @SuppressWarnings("UnstableApiUsage")
     private RetryerBuilder<List<IndexingError>> createBulkRequestRetryerBuilder() {
         return RetryerBuilder.<List<IndexingError>>newBuilder()
-                .retryIfException(t -> t instanceof IOException || t instanceof InvalidWriteTargetException)
+                .retryIfException(t -> t instanceof IOException || t instanceof InvalidWriteTargetException || t.getCause().getCause() instanceof IOException)
```

Comment:
Since `t.getCause()` can return `null`, I think we should use `ExceptionUtils.getRootCause()` to make sure we don't throw a `NullPointerException`. The `IOException` also might be deeper than two levels.

```diff
--- graylog2-server/src/main/java/org/graylog2/indexer/messages/Messages.java
+++ graylog2-server/src/main/java/org/graylog2/indexer/messages/Messages.java
@@ -53,6 +53,8 @@ import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
+import static org.graylog2.shared.utilities.ExceptionUtils.getRootCause;
+
 @Singleton
 public class Messages {
     public interface IndexingListener {
@@ -78,7 +80,7 @@ public class Messages {
     @SuppressWarnings("UnstableApiUsage")
     private RetryerBuilder<List<IndexingError>> createBulkRequestRetryerBuilder() {
         return RetryerBuilder.<List<IndexingError>>newBuilder()
-                .retryIfException(t -> t instanceof IOException || t instanceof InvalidWriteTargetException)
+                .retryIfException(t -> t instanceof IOException || getRootCause(t) instanceof IOException || t instanceof InvalidWriteTargetException)
                 .withWaitStrategy(WaitStrategies.exponentialWait(MAX_WAIT_TIME.getQuantity(), MAX_WAIT_TIME.getUnit()))
                 .withRetryListener(new RetryListener() {
                     @Override
```